mtp_transaction_uploader/upload.py

```diff
@@ -0,0 +1,139 @@
+import os
+import shutil
+import re
+from datetime import datetime
+from collections import namedtuple
+
+from bankline_parser.data_services import parse_file
+from bankline_parser.data_services.enums import TransactionCode
+from pysftp import Connection
+
+from .api_client import get_authenticated_connection
+from . import settings
+
+DATE_FORMAT = '%d%m%y'
+
+ref_pattern = re.compile('''
+    ([A-Za-z][0-9]{4}[A-Za-z]{2}) # match the prisoner number
+    .*?                           # skip until next digit
+    ([0-9]{1,2})                  # match 1 or 2 digit day of month
+    .*?                           # skip until next digit
+    ([0-9]{1,2})                  # match 1 or 2 digit month
+    .*?                           # skip until next digit
+    ([0-9]{2,4})                  # match 2 or 4 digit year
+    ''', re.X)
+file_pattern = re.compile('''
+    YO1A\.REC\.#D\.               # static file format
+    (%(code)s)\.                  # our unique account code
+    D([0-9]{6})                   # date that file was generated (ddmmyy)
+    ''' % {'code': settings.ACCOUNT_CODE}, re.X)
+
+
+def retrieve_data_services_files():
+    # check for existing downloaded files and remove if found
+    if os.path.exists(settings.DS_NEW_FILES_DIR):
+        shutil.rmtree(settings.DS_NEW_FILES_DIR)
+    os.mkdir(settings.DS_NEW_FILES_DIR)
+
+    # check if we recorded date of last file downloaded
+    last_date = None
```

Comment:
Would be worth setting a default, a value really far in the past. This is because the check https://github.com/ministryofjustice/money-to-prisoners-transaction-uploader/pull/1/files#diff-8d4a24f3bb40368c078fa6bfdde77673R56 wouldn't work properly otherwise I think